stellargraph/core/graph.py

```diff
@@ -433,17 +712,54 @@ def to_adjacency_matrix(self, nodes: Optional[Iterable] = None):
         Returns:
              The weighted adjacency matrix.
         """
-        return self._graph.to_adjacency_matrix(nodes)
-
-    def to_networkx(self):
+        if self._graph is not None:
+            return self._graph.to_adjacency_matrix(nodes)
+
+        if nodes is None:
+            index = self._nodes._id_index
+            selector = slice(None)
+        else:
+            nodes = list(nodes)
+            index = ExternalIdIndex(nodes)
+            selector = self._edges.sources.isin(nodes) & self._edges.targets.isin(nodes)
+
+        weights = self._edges.weights[selector]
+        src_idx = index.to_iloc(self._edges.sources[selector])
+        tgt_idx = index.to_iloc(self._edges.targets[selector])
```

Comment:
Hmm I think when `nodes is not None` (i.e. a subgraph), the ilocs here need to be reassigned in some way? Otherwise they won't correspond with the shape `n = len(index)` but instead with the full graph.

Reply:
They are being reassigned by the newly created `ExternalIdIndex`; that ensures they're relative to the `nodes` list. I'll add a comment.